transaction-coordinator/src/main/java/org/apache/kafka/coordinator/transaction/TransactionState.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.transaction;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the states of a transaction in the transaction coordinator.
+ * This enum corresponds to the Scala sealed trait TransactionState in kafka.coordinator.transaction.
+ */
+public enum TransactionState {
+    /**
+     * Transaction has not existed yet
+     * <p>
+     * transition: received AddPartitionsToTxnRequest => Ongoing
+     *             received AddOffsetsToTxnRequest => Ongoing
+     *             received EndTxnRequest with abort and TransactionV2 enabled => PrepareAbort
+     */
+    EMPTY((byte) 0, "Empty", true),
```

Comment:
the name is exposed by the RPC, so it would be good to ensure the naming between this and admin.TransactionState is consistent. Maybe we can use the following pattern to rewrite it?
```java
EMPTY((byte) 0, org.apache.kafka.clients.admin.TransactionState.EMPTY.toString(), true),
```